### PR TITLE
redo(ticdc):  fix resolvedTs and split transaction in applier

### LINKED ISSUE #8318: redo applier ignores some data
### What did you do?

1. run sysbench to prepare 10 million rows
2. create changefeed with redo enabled, and sleep for 150 seconds
3. pause changefeed
4. run redo apply and check data consistency

### What did you expect to see?

Pass consistency check

### What did you see instead?
Some data is lost in the downstream database:
```log
The table structure and data in following tables are equivalent

+-----------------------+---------+-----------+
|         TABLE         | UPCOUNT | DOWNCOUNT |
+-----------------------+---------+-----------+
| `workload`.`sbtest10` |  790272 |    790272 |
| `workload`.`sbtest1`  |  827904 |    827904 |
| `workload`.`sbtest2`  |  798336 |    798336 |
| `workload`.`sbtest3`  |  790272 |    790272 |
| `workload`.`sbtest4`  |  787584 |    787584 |
| `workload`.`sbtest6`  |  798336 |    798336 |
| `workload`.`sbtest7`  |  827904 |    827904 |
| `workload`.`sbtest8`  |  809088 |    809088 |
| `workload`.`sbtest9`  |  795648 |    795648 |
+-----------------------+---------+-----------+


The following tables contains inconsistent data

+----------------------+--------------------+----------------+---------+-----------+
|        TABLE         | STRUCTURE EQUALITY | DATA DIFF ROWS | UPCOUNT | DOWNCOUNT |
+----------------------+--------------------+----------------+---------+-----------+
| `workload`.`sbtest5` | true               | +256/-0        |  782208 |    781952 |
+----------------------+--------------------+----------------+---------+-----------+

```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
(paste TiCDC version here)
```

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #8318 

### What is changed and how it works?
1.  fix resolvedTs and split transaction in applier

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
`Split transaction in applier to reduce RTO`.
```

### CODE REVIEW

File: pkg/applier/redo.go [hunk 1]
```diff
@@ -145,41 +144,33 @@ func (ra *RedoApplier) consumeLogs(ctx context.Context) error {
...
 			if redoLog.CommitTs > tableResolvedTsMap[tableID] {
 				tableResolvedTsMap[tableID] = redoLog.CommitTs
 			}
 		}
 
-		for tableID, tableLastResolvedTs := range tableResolvedTsMap {
-			if err := ra.tableSinks[tableID].UpdateResolvedTs(
-				model.NewResolvedTs(tableLastResolvedTs)); err != nil {
+		for tableID, tableResolvedTs := range tableResolvedTsMap {
+			batchRts := model.ResolvedTs{
```
Comment (hi-rustin) [inline pkg/applier/redo.go:156]: Better to add a comment to explain why we need a batch resolved ts here.

File: pkg/applier/redo.go [hunk 2]
```diff
@@ -145,41 +144,33 @@ func (ra *RedoApplier) consumeLogs(ctx context.Context) error {
 				ra.tableSinks[tableID] = tableSink
 			}
 			if _, ok := tableResolvedTsMap[tableID]; !ok {
-				// Safe to use checkpointTs - 1 as the resolvedTs of a table.
-				tableResolvedTsMap[tableID] = checkpointTs - 1
+				tableResolvedTsMap[tableID] = checkpointTs
```
Comment (hi-rustin) [inline pkg/applier/redo.go:147]: ditto

**Comment by hi-rustin** [comment]
Any possible to add a test to cover this case?

**Comment by CharlesCheung96** [comment]
> Any possible to add a test to cover this case?

Currently we need the test_infra test to ensure correctness, and we may add some unit tests in anthor pr later.

**Review by hi-rustin — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**